### PR TITLE
feat(codepipeline-actions): support region attribute on CodeBuildAction

### LINKED ISSUE #10279: [aws-codepipeline-actions] codebuild cross-region build actions
AWS CodePipeline [Now Supports Cross-Region Actions](https://aws.amazon.com/about-aws/whats-new/2018/11/aws-codepipeline-now-supports-cross-region-actions/) and the CDK has been updated to allow cross-region support for cloudformation but has not been added for CodeBuild.

In https://docs.aws.amazon.com/cdk/api/latest/docs/aws-codepipeline-readme.html CloudFormationCreateUpdateStackAction is referenced as a cross-region action but attempting to use the same implementation for a CodeBuild project results in 

> Object literal may only specify known properties, and 'region' does not exist

### Use Case

I would like to be able to provision CodePipelines in CDK that utilizes cross-region CodeBuild projects. I have my primary CodeCommit repository in RegionA but have some build actions that can only be run in RegionB due to network connectivity/latency.

This capability is available via the console and CloudFormation but hasn't been introduced to CDK.

### Proposed Solution

I have a PR in waiting that, making a similar change as was implemented in https://github.com/aws/aws-cdk/issues/49 allowing for CodeBuildAction to support the region attribute.

### Other

Below is an example stack that shows current and proposed functionality.

```typescript
import {Construct, Stack, StackProps} from '@aws-cdk/core'
import {PipelineProject} from '@aws-cdk/aws-codebuild'
import {Repository} from '@aws-cdk/aws-codecommit'
import {Artifact, Pipeline} from '@aws-cdk/aws-codepipeline'
import {CodeBuildAction, CodeCommitSourceAction} from '@aws-cdk/aws-codepipeline-actions'
import {Bucket, IBucket} from '@aws-cdk/aws-s3'

export class PipelineStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props)

    const appName = this.node.tryGetContext('appName')
    const regionSecondary = this.node.tryGetContext('regionSecondary')

    const artifactBucket = new Bucket(this, 'PipelineBucket', {
      bucketName: `${appName}-${this.region}-${this.account}`,
    })

    const crossRegionBuckets: {[key: string]: IBucket} = {}

    crossRegionBuckets[this.region] = artifactBucket
    crossRegionBuckets[regionSecondary] = Bucket.fromBucketName(this, 'CrossRegionBucket',
      `${appName}-${regionSecondary}-${this.account}`,
    )

    const sourceArtifact = new Artifact('CodeCommitSource')
    const sourceAction = new CodeCommitSourceAction({
      actionName: 'CodeCommit',
      repository: Repository.fromRepositoryArn(this, 'CodeRepository',
        this.node.tryGetContext('existingRepositoryArn'),
      ),
      output: sourceArtifact,
    })

    // this is valid
    const sameRegionBuildAction = new CodeBuildAction({
      actionName: 'CodeBuild',
      project: new PipelineProject(this, 'SameRegionBuildProject'),
      input: sourceArtifact,
    })

    // this is not invalid
    const crossRegionBuildAction = new CodeBuildAction({
      actionName: 'CodeBuild',
      project: PipelineProject.fromProjectName(this, 'CrossRegionBuildProject',
        this.node.tryGetContext('crossRegionBuildProjectArn'),
      ),
      input: sourceArtifact,
      region: regionSecondary,
    })

    new Pipeline(this, 'CodePipeline', {
      crossRegionReplicationBuckets: crossRegionBuckets,
      stages: [
        {stageName: 'Source', actions: [sourceAction]},
        {stageName: 'Build', actions: [sameRegionBuildAction, crossRegionBuildAction]},
      ],
    })
  }
}
```

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by amhammett**
fixes #10279

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-codepipeline-actions/README.md [hunk 1]
```diff
@@ -321,6 +321,24 @@ const testAction = new codepipeline_actions.CodeBuildAction({
 });
 ```
 
+#### Cross-Region CodeBuild Actions
```
Comment (amhammett) [inline packages/@aws-cdk/aws-codepipeline-actions/README.md:324]: I have tried to make this section as concise as possible to show what is required to implement cross-region build actions. I'm happy to expand this.

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the PR @amhammett , but this is a non-starter. Actions that are backed by a resource take its region from the resource's region, you cannot provide one separately (it doesn't make much sense).

In your case, you probably want to the correct logic in `fromProjectArn(): IProject` method in the `Project` class to take the region out of the ARN, very similarly like it [works for Bucket today](https://github.com/aws/aws-cdk/blob/287f808cf34870295e4032995fc083ac204b64c3/packages/@aws-cdk/aws-codepipeline-actions/README.md#aws-s3).

**Comment by amhammett** [comment]
Thanks for the feedback @skinny85 I've taken a look at the code you referenced and just need to get my head around and finish some testing before I update this PR.

**Comment by skinny85** [comment]
> Thanks for the feedback @skinny85 I've taken a look at the code you referenced and just need to get my head around and finish some testing before I update this PR.

Sounds good, let me know if you need any additional information, I'll be happy to assist 🙂.

**Comment by skinny85** [comment]
I'm closing this one due to inactivity. Please comment if you want it re-opened.

**Pull request closed without merging**